sdk/core/Azure.Core.TestFramework/src/TestRecordingMismatchException.cs

```diff
@@ -21,10 +21,6 @@ public TestRecordingMismatchException(string message, Exception innerException)
         {
         }
 
-        protected TestRecordingMismatchException(SerializationInfo info, StreamingContext context) : base(info, context)
```

Comment:
This is generally a standard part of a custom exception.   Did something change in net8.0 that causes us to remove it?

Reply:
This specific overload is now obsolete and won't compile anymore in .NET 8

![image](https://github.com/user-attachments/assets/65928739-516e-4454-9a5d-008a721be00f)

Reply:
https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051

Reply:
Ok, so this was a hook for the binary formatter.   I did not know that!